transactions.c

```diff
@@ -179,6 +179,40 @@ void returnBook(int book_uid)
     printf("Thank you for choosing LMS!\n--------------------\n");
 }
 
+void purchaseBook(Book* book, User* user)
+{
+    // Check if source book has available copies in stock
+    if (book->in_stock_count < 1 || book->status != active)
+    {
+        printf("Purchase unsuccessful. No copy available for purchase\n--------------\n");
+        return;
+    }
+
+    // Get the first available copy
+    BookCopy* purchased_copy = getBookCopy(book);
```

Comment:
Handle edge (null) case

Reply:
I just rechecked the code. Before like 190 was the code to check if any copies available: 
(line 184-189)
// Check if source book has available copies in stock
    if (book->in_stock_count < 1 || book->status != active)
    {
        printf("Purchase unsuccessful. No copy available for purchase\n--------------\n");
        return;
    }
so if there wasn't any copy available for rent, this function would already be returned before getting to line 192.